### PR TITLE
Admin feature: adding multiple users from file.

### PR DESCRIPTION
**Pull request by JanCBrammer**
# Description

Currently, admins can only add one user at a time by submitting a form:
add_single_user

This PR adds a feature that allows admins to add multiple users from a CSV file:
add_multiple_users

The feature has been requested by @alhoch.

# New dependencies
- [react-papaparse](https://react-papaparse.js.org/)

# Checklist
- [x] added feature to CHANGELOG
- [x] added acceptance test (passing locally and on CI)

### CODE REVIEW

File: spec/features/user_management_spec.rb [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  end
+
+  def write_missing_cell
+    invalid_users = @users.dup
+    invalid_users[rand(invalid_users.size)].pop
+    write(@header, invalid_users)
+  end
+end
+
+PROCESSING_SUMMARY = # FIXME: string formatting might be brittle, i.e., matches might depend on runtime environment.
```
Comment (PiTrem) [inline spec/features/user_management_spec.rb:69]: ( squiggly heredoc ? )

File: app/packs/src/admin/UserManagement.js [hunk 2]
```diff
@@ -256,6 +259,143 @@ export default class UserManagement extends React.Component {
...
+    if (!validFileTypes.includes(file.type)) { // Note that MIME type doesn't reliably indicate file type. It's only an initial guard and data is validated more thoroughly during processing.
+      this.setState({ userFileProcessingSummary: `Invalid file type ${file.type}. Please make sure to upload a CSV file.` });
+      this.newUsers = null;
+      return false;
+    }
+    this.newUsers = data;
+    for (let i = 0; i < this.newUsers.length; i++) {
+      this.newUsers[i].data.row = i + 1;
+    }
+    console.log(this.newUsers);
```
Comment (PiTrem) [inline app/packs/src/admin/UserManagement.js:273]: to be removed

File: app/packs/src/admin/UserManagement.js [hunk 3]
```diff
@@ -256,6 +259,143 @@ export default class UserManagement extends React.Component {
...
+  };
+
+  handleOnErrorUserFile = (err, file, inputElem, reason) => {
+    console.log(err);
+    this.newUsers = null;
+  };
+
+  handleOnRemoveUserFile = (data) => {
+    console.log(data);
```
Comment (PiTrem) [inline app/packs/src/admin/UserManagement.js:282]: to be removed

File: app/packs/src/admin/UserManagement.js [hunk 4]
```diff
@@ -256,6 +259,143 @@ export default class UserManagement extends React.Component {
...
+  handleCreateNewUsersFromFile() {
+    if (!this.validUserFile()) {
+      this.newUsers = null;
+      this.setState({ createUserMessage: 'Finished processing user file.' });
+    } else {
+      const promisedNewUsers = this.newUsers.map(user => this.createNewUserFromFile(user));
+      Promise.allSettled(promisedNewUsers)
+        .then((userResults) => {
+          this.setState({ createUserMessage: 'Finished processing user file.' });
+          this.showProcessingSummaryUserFile(userResults);
```
Comment (PiTrem) [inline app/packs/src/admin/UserManagement.js:295]: showProcessingSummaryUserFile is calling setState too. basically you have 2 setState in a row with each probably going to trigger rendering. (also not highly likely but the 2nd one could finish before the 1st one )
could you refactor using one  setState ?

File: spec/features/user_management_spec.rb (hunk 1 above)
Comment (JanCBrammer) [reply spec/features/user_management_spec.rb]: See 67a0899.

File: app/packs/src/admin/UserManagement.js (hunk 2 above)
Comment (JanCBrammer) [reply app/packs/src/admin/UserManagement.js]: See d680d99.

File: app/packs/src/admin/UserManagement.js (hunk 3 above)
Comment (JanCBrammer) [reply app/packs/src/admin/UserManagement.js]: See d680d99.

File: app/packs/src/admin/UserManagement.js (hunk 4 above)
Comment (JanCBrammer) [reply app/packs/src/admin/UserManagement.js:292]: See e085070.

**Comment by JanCBrammer** [comment]
@PiTrem, I addressed your comments above. If you're ok with the changes, I'll edit the commit history to be more comprehensive as discussed (basically one commit for the new feature, one commit for the tests, one commit for CHANGELOG).

**Pull request merged**